src/components/common/Dialogs/CreatePaymentDialog/NetworkFee/NetworkFee.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import {
+  defineMessages,
+  FormattedMessage,
+  MessageDescriptor,
+} from 'react-intl';
+import { useFormContext } from 'react-hook-form';
+import Decimal from 'decimal.js';
+
+import { calculateFee, getTokenDecimalsWithFallback } from '~utils/tokens';
+import { notNull } from '~utils/arrays';
+import Numeral from '~shared/Numeral';
+import { Colony } from '~types';
+
+import styles from './NetworkFee.css';
+
+const displayName =
+  'common.CreatePaymentDialog.CreatePaymentDialogForm.NetworkFee';
+
+const MSG = defineMessages({
+  amount: {
+    id: `${displayName}.amount`,
+    defaultMessage: 'Amount',
+  },
+  fee: {
+    id: `${displayName}.fee`,
+    defaultMessage: 'Network fee: {fee} {symbol}',
+  },
+  token: {
+    id: `${displayName}.address`,
+    defaultMessage: 'Token',
+  },
+});
+
+interface Props {
+  colony: Colony | undefined;
+  customAmountError: string | MessageDescriptor | undefined;
+  networkFeeInverse: string | undefined;
+}
+
+const NetworkFee = ({
+  colony,
+  networkFeeInverse,
+  customAmountError,
+}: Props) => {
+  const { getValues } = useFormContext();
+  const values = getValues();
+  const colonyTokens =
+    colony?.tokens?.items
+      .filter(notNull)
+      .map((colonyToken) => colonyToken.token) || [];
+  const selectedToken = colonyTokens.find(
+    (token) => token?.tokenAddress === values.tokenAddress,
+  );
+
+  if (
+    !networkFeeInverse &&
+    customAmountError &&
+    new Decimal(values.amount).isZero()
+  ) {
+    return null;
+  }
+
+  return (
+    <div className={styles.networkFee}>
+      <FormattedMessage
+        {...MSG.fee}
+        values={{
+          fee: (
+            <Numeral
+              // appearance={{
+              //   size: 'small',
+              //   theme: 'grey',
+              // }}
+              value={
+                calculateFee(
+                  values.amount,
+                  networkFeeInverse!,
```

Comment:
This triggers eslint and actually looks like unsafe assertion. This prop can be undefined and you don't have any checks that would ensure it isn't

Reply:
But we do have checks, before this render:
```
  if (
    !networkFeeInverse &&
    customAmountError &&
    new Decimal(values.amount).isZero()
  ) {
    return null;
  }
```

But I'll think about implementing something clearer to eslint